### PR TITLE
Patches #35 - op_endpoint and identifier validation

### PR DESCRIPTION
**Pull request by kondax**
Fixes non-validation issue in bug #35. Should be temporary, but with the amount of people just copying the example, this needs to be set.

### CODE REVIEW

File: examples/signon/app.js [hunk 1]
```diff
@@ -26,9 +26,14 @@ passport.deserializeUser(function(obj, done) {
 passport.use(new SteamStrategy({
     returnURL: 'http://localhost:3000/auth/steam/return',
     realm: 'http://localhost:3000/',
-    apiKey: 'Your API key here'
+    apiKey: 'Your API Key Here',
+    //passReqToCallback -- now deprecated, since request is always returned.
```
Comment (scholtzm) [inline examples/signon/app.js]: Is it? https://github.com/jaredhanson/passport-openid/blob/master/lib/passport-openid/strategy.js#L41

File: lib/passport-steam/strategy.js [hunk 2]
```diff
@@ -47,12 +47,12 @@ function Strategy(options, validate) {
 
   if(options.profile) {
     var steam = new SteamWebAPI({ apiKey: options.apiKey, format: 'json' });
-
+
```
Comment (scholtzm) [inline lib/passport-steam/strategy.js]: Extraneous whitespace.

File: lib/passport-steam/strategy.js [hunk 3]
```diff
@@ -47,12 +47,12 @@ function Strategy(options, validate) {
...
   if(options.profile) {
     var steam = new SteamWebAPI({ apiKey: options.apiKey, format: 'json' });
-
+    
     function getUserProfile() {
-      var req = arguments[options.passReqToCallback ? 0 : undefined];
-      var identifier = arguments[options.passReqToCallback ? 1 : 0];
-      var profile = arguments[options.passReqToCallback ? 2 : 1];
-      var done = arguments[options.passReqToCallback ? 3 : 2];
+      var req = arguments[0];
```
Comment (scholtzm) [inline lib/passport-steam/strategy.js]: No need for `arguments` if their count is constant.

File: examples/signon/app.js (hunk 1 above)
Comment (kondax) [inline examples/signon/app.js]: I mean in terms for this library, as a temporary fix.

**Comment by kondax** [comment]
@scholtzm Acted upon your comments.

File: lib/passport-steam/strategy.js [hunk 4]
```diff
@@ -49,11 +49,6 @@ function Strategy(options, validate) {
     var steam = new SteamWebAPI({ apiKey: options.apiKey, format: 'json' });
 
     function getUserProfile() {
-      var req = arguments[options.passReqToCallback ? 0 : undefined];
-      var identifier = arguments[options.passReqToCallback ? 1 : 0];
-      var profile = arguments[options.passReqToCallback ? 2 : 1];
-      var done = arguments[options.passReqToCallback ? 3 : 2];
-
       steam.getPlayerSummaries({
         steamids: [ identifier ],
```
Comment (scholtzm) [inline lib/passport-steam/strategy.js]: But now `identifier` is undefined and that's not what I meant. Just add the arguments to function signature like so: `function getUserProfile(req, identifier, profile, done)`

Comment (kondax) [inline lib/passport-steam/strategy.js]: Just setting `var identifier = arguments[1]` would be enough.

Comment (scholtzm) [inline lib/passport-steam/strategy.js]: Hiding arguments makes no sense but not going to argue.

Comment (kondax) [inline lib/passport-steam/strategy.js]: Well, not needed anywhere else.

**Comment by scholtzm** [comment]
Sanity checks should be done outside of `getUserProfile`. [This patch still fails to validate the identifier if `options.profile` is `falsy`.](https://github.com/kondax/passport-steam/blob/e754ccdd7a037760640368e23cb5e6cfc45e9b1d/lib/passport-steam/strategy.js#L75)

**Comment by kondax** [comment]
@scholtzm Check the latest commit. That should be valid, but I'm not aware of the OpenID Spec so do note if it isn't.

File: lib/passport-steam/strategy.js [hunk 5]
```diff
@@ -41,23 +41,23 @@ var util = require('util')
  */
 function Strategy(options, validate) {
   options = options || {};
-  options.providerURL = options.providerURL || 'http://steamcommunity.com/openid';
-  options.profile =  (options.profile === undefined) ? true : options.profile;
+  options.providerURL = 'http://steamcommunity.com/openid';
+  options.profile =  options.profile || true;
   options.stateless = true; //Steam only works as a stateless OpenID
 
+  if(typeof arguments[0].query !== 'undefined' && arguments[0].query['openid.op_endpoint'] !== 'https://steamcommunity.com/openid/login')
```
Comment (scholtzm) [inline lib/passport-steam/strategy.js]: This won't work.

Comment (kondax) [inline lib/passport-steam/strategy.js]: I've got no idea then. I didn't think it would.

**Pull request closed without merging**